### ISSUE TITLE
DEPR: Deprecate xlrd and pyxlsb excel engines

### ISSUE THREAD

**Issue by phofl** [pandas-dev/pandas#56542]
We added calamine support for 2.2, so we can theoretically deprecate these 2 engines in favour of calamine to reduce our dependencies and just code that isn't needed anymore

**Comment by rhshadrach** [comment]
+1

In general I'm okay with supporting any "good enough" engine even if it means we have multiple engines for certain formats, but neither of these fit that criteria in my opinion. I would wait a bit for users to give feedback on calamine though. Also might think about deprecating the default to become calamine first, then deprecating these altogether?

**Comment by asishm** [comment]
I like the idea of making calamine the default first and then deprecating (similar transition had taken place when the default engine was changed for xlsx files)

**Comment by jbrockmendel** [comment]
For my edification, is calamine better in some way than the others?  IIRC xlrd is not maintained; is that accurate?

**Comment by rhshadrach** [comment]
> xlrd is not maintained; is that accurate?

Yes - and for pyxlsb, it doesn't offer some features that I would consider basic, like [reading timestamps](https://github.com/pandas-dev/pandas/blob/ac170fdc3570b55bfcaf44b8587825bcb274bc60/pandas/tests/io/excel/test_readers.py#L146).

**Comment by jbrockmendel** [comment]
Time to move forward with this?

**Comment by rhshadrach** [comment]
Yep I think so. I think I should have time to take this up.
